web/src/app/modules/shared/components/presentation/icon/icon.component.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+} from '@angular/core';
+import { AbstractViewComponent } from '../../abstract-view/abstract-view.component';
+import { IconView } from '../../../models/content';
+
+@Component({
+  selector: 'app-view-icon',
+  templateUrl: './icon.component.html',
+  styleUrls: ['./icon.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IconComponent extends AbstractViewComponent<IconView> {
+  shape: string;
+  flip: string;
+  size: string;
+
+  styleAttr = '';
+  classAttr = '';
+
+  sizeHash = { sm: '16', md: '24', lg: '36"', xl: '48', xxl: '64' };
+  directionHash = {
```

Comment:
After this merges, please create an issue that references all of these locations we can remove these mappings once we migrate from `clr-icon` to `cds-icon`

Reply:
Sure!